pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -337,6 +337,46 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env model.Env
 	return listeners
 }
 
+type listenerEntry struct {
+	service *model.Service
+	servicePort *model.Port
+	listener *xdsapi.Listener
+}
+
+func getServiceName(env model.Environment, service *model.Service) string {
+	attrs, err := env.GetServiceAttributes(service.Hostname)
+	if err != nil || attrs == nil || attrs.Name == "" {
+		return "unknown"
+	}
+	return fmt.Sprintf("%s/%s", attrs.Namespace, attrs.Name)
+}
+
+func getClusterName(service *model.Service, servicePort *model.Port) string {
+	return model.BuildSubsetKey(model.TrafficDirectionOutbound, "",
+		service.Hostname, servicePort.Port)
+}
+
+func handleOutboundListenerConflict(env model.Environment, service *model.Service, servicePort *model.Port, current listenerEntry, listenerMapKey string) {
+	conflictingOutbound.Add(1)
+	log.Warnf("buildSidecarOutboundListener: Omitting listener for service '%s' due to conflict with service '%s'. "+
```

Comment:
This is likely to go away and be replaced by the metric, this kind of log is hard to track in pilot logs due to noise (each log is repeated for every single sidecar, along with lots of other logs )